server/proxy/notify.go

```diff
@@ -15,167 +17,199 @@ import (
 // NotifyUserCreated handles plugin's UserHasBeenCreated callback. It emits
 // "user_created" notifications to subscribed apps.
 func (p *Proxy) NotifyUserCreated(userID string) {
-	p.notify(nil,
+	p.notifyAll(
 		apps.Event{
 			Subject: apps.SubjectUserCreated,
 		},
 		apps.UserAgentContext{
 			UserID: userID,
 		},
+		nil, // no special filtering, notify all subscriptions mathcing the event.
```

Comment:
I think we'd need a Notification struct for the future so we can remove these comments and have less parameters on some of the methods.